File: order-mgmt/src/main/kotlin/com/order/ordermgmt/models/StationOrder.kt
package com.order.ordermgmt.models

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.time.LocalDateTime
import javax.persistence.*

@Entity
class StationOrder(
        val status: OrderStatus,
        @ManyToOne
        @JoinColumn(name="station_id")
        val station: Station,

        @ManyToOne
        @JoinColumn
        val clientOrder: ClientOrder,

        @ManyToOne
        @JoinColumn(name="cook_id")
        val cook: Cook?

):BaseEntity(){
    var completionTime:Int=0
    @ManyToMany()
    @JoinTable(name="station_order_dish",
            joinColumns = [(JoinColumn(name="order_id", referencedColumnName = "id"))],
            inverseJoinColumns = [(JoinColumn(name = "dish_id",referencedColumnName = "id"))])
    @JsonIgnore
    val dishList= mutableListOf<Dish>()


    fun addDishToOrder(dish:Dish){
        completionTime+=dish.cookTime
        dishList.add(dish)
    }

    fun addDishListToOrder(dishList:MutableList<Dish>):StationOrder{
        dishList.forEach {
            completionTime+=it.cookTime
            dishList.add(it)
        }
        return this
    }
}

File: restaurant_app/lib/models/station_order.dart
import 'package:common_widgets/models/dish.dart';
import 'package:meta/meta.dart';

class StationOrder{
  int id;
  int stationId;
  int clientOrderId;
  int completionTime;
  List<Dish> dishList;

  StationOrder({
    @required this.id,
    @required this.stationId,
    @required this.clientOrderId,
    @required this.completionTime,
    @required this.dishList
  });

  static StationOrder fromMap(Map<String,dynamic> stationOrderMap){
    int compltime=0;
    (stationOrderMap["dishList"] as List).forEach((dish)=> compltime+=dish.cookTime);
    return StationOrder(
      id: stationOrderMap["id"],
      stationId: stationOrderMap["station"]["id"],
      clientOrderId: stationOrderMap["clientOrder"]["id"],
      completionTime: compltime,
      dishList: (stationOrderMap["dishList"] as List).map((dishMap)=>Dish.fromMap(dishMap)).toList()
      );
  }
}